pkg/containerutil/container_network_manager_windows.go

```diff
@@ -0,0 +1,193 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package containerutil
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/oci"
+	"github.com/containerd/containerd/pkg/netns"
+	gocni "github.com/containerd/go-cni"
+
+	"github.com/containerd/nerdctl/pkg/api/types"
+	"github.com/containerd/nerdctl/pkg/netutil"
+	"github.com/containerd/nerdctl/pkg/ocihook"
+)
+
+// Verifies that the internal network settings are correct.
+func (m *cniNetworkManager) VerifyNetworkOptions(_ context.Context) error {
+	e, err := netutil.NewCNIEnv(m.globalOptions.CNIPath, m.globalOptions.CNINetConfPath, netutil.WithDefaultNetwork())
+	if err != nil {
+		return err
+	}
+
+	// NOTE: only currently supported network type on Windows is nat:
```

Comment:
why is only nat supported?

Reply:
The main goal of the patch was to get minimal CNI networking running on Windows, enable as many tests which use networking (e.g. tests which set up a temporary local registry), and then add other network modes later.

Seeing as though `nat` was the [default network type set up by the CI](https://github.com/containerd/nerdctl/blob/main/hack/configure-windows-ci.ps1#L31), and is the only one that gets tested, I added this explicit constraint for now.